### DISCUSSION TITLE
Secure connection

### DISCUSSION THREAD

**Discussion by stefanomologni** [Infineon/optiga-trust-m#60] [Q&A]
Dear all,
  I'm not sure about using Secure Connection with Optiga TRUST M library on EIA 232 chip.

Starting from scratch I'll share a secret with OPTIGA using datastore, during first boot. After that I can set a protection level and execute some operations. After that, if i restart OPTIGA chip, the protection level is manteined? The communication still secure?

After this reboot the host can access to data into EIA 232 without the shared secret?

There are some examples of code using this feature?

Best regards,
Stefano

**Comment by ayushev** [comment]
Sorry for a late reply.
I think in general you have correctly understood the flow, there are some details which I would like to clarify:
1) The Host and the Secure Element maintains a communication session, it is typcially stored in the RAM like memmory
2) It means with a lost of power on the host side, you loos the communication session and once powered back both sides need to re-establish it (it takes some time and also triggers a security event on the SE side as this is a symmetric key operation performed on the chip)
3) If you would like to avoid this, you can bring the chip into y hibernation mode right before power down.

> After this reboot the host can access to data into EIA 232 without the shared secret?

I think this a bit different question, as Data Ibject Access Conditions define whetehr you need a shared secret to access the data. For example you can read out a certificate object for generic samples without any secured channel established.

> There are some examples of code using this feature?

I think we can find a code example for nmany things, could you please specify more precisely what would you like to see (if this is still relevant for you)?
